### DISCUSSION TITLE
Image and Blob search

### DISCUSSION THREAD

**Discussion by dberardo-com** [duckdb/duckdb#10761] [Ideas]
Is it possible to use duckdb to perform some Blob / Object search on an S3 bucket based on names / size / date of creation or even tags, and other metadata (when available) ?

**Comment by carlopi** [comment]
There is read_blob: https://duckdb.org/docs/guides/import/read_file#read_blob.

Note that if you do `SELECT * EXCEPT(content) FROM read_blob('s3://bucket-name/**/*.jpg')` the actual content is NOT retrieved, so it's way faster given only metadata is read.

**Reply by dberardo-com** [reply]
thanks for that, i was not able to find that out easily from the doc.

i will give it a try and let you know if that covers my use case.
